src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -173,10 +173,14 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            // restart the timer with the next schedule occurrence
             await InvokeJobFunction(DateTime.Now, false);
 
-            StartTimer(DateTime.Now);
+            // Restart the timer with the next schedule occurrence, but only 
+            // if Cancel, Stop, and Dispose have not been called.
+            if (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                StartTimer(DateTime.Now);
```

Comment:
I'm looking at a similar bug with DateTime right now and so this caught my eye. Would .UtcNow be more correct here?   .Now is timezone specific and can have oddities in different timezones and around daylight savings. We should always use Utc (timezone agnostic)